app/src/main/java/ai/brokk/http/HttpServer.java

```diff
@@ -0,0 +1,472 @@
+
+            String body = req.getReader().lines().collect(Collectors.joining("\n"));
+            if (body == null || body.isBlank()) {
+                resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                resp.getWriter().println("{\"error\":\"Empty request body\"}");
+                return;
+            }
+
+            Map<String, Object> prData;
+            try {
+                prData = ai.brokk.cli.BrokkCli.parseReviewInput(body);
+            } catch (Exception e) {
+                resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                resp.getWriter().println(mapper.writeValueAsString(Map.of(
+                        "error", "Invalid JSON payload",
+                        "detail", e.getMessage())));
+                return;
+            }
+
+            if (!ai.brokk.cli.BrokkCli.validatePrData(prData)) {
+                resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                resp.getWriter().println(mapper.writeValueAsString(Map.of(
+                        "error", "PR data is missing required fields",
+                        "required", "pr_number, title, author, base_branch, head_branch, changed_files")));
+                return;
+            }
+
+            synchronized (BROKK_KEY_LOCK) {
+                String prevKey = MainProject.getBrokkKey();
+                Path tempProjectPath = null;
+                ContextManager cm = null;
+
+                try {
+                    MainProject.setBrokkKey(apiKey.trim());
+
+                    tempProjectPath = Files.createTempDirectory("brokk-review-");
+                    var mainProject = new MainProject(tempProjectPath);
+                    var project = mainProject;
+                    cm = new ContextManager(project);
+
+                    cm.createHeadless();
+                    cm.replaceIo(new ai.brokk.cli.SilentConsole());
+
+                    Service.FavoriteModel favModel;
+                    try {
+                        favModel = MainProject.getFavoriteModel(codeModelAlias);
+                    } catch (IllegalArgumentException e) {
+                        resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                        resp.getWriter().println(mapper.writeValueAsString(Map.of(
+                                "error", "Unknown code model",
+                                "model", codeModelAlias)));
+                        return;
+                    }
+
+                    var codeModel = cm.getService().getModel(favModel.config());
+                    if (codeModel == null) {
+                        resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                        resp.getWriter().println(mapper.writeValueAsString(Map.of(
+                                "error", "Failed to initialize code model")));
+                        return;
+                    }
+
+                    if (sessionIdStr != null && !sessionIdStr.isBlank()) {
```

Comment:
I don't think we'll want all this code repeated. Is there something I'm missing between this and my other comments?